### PR TITLE
build a magisk module as well; also some minor fixes:

### PR DESCRIPTION
**Pull request by infinity0**
- it's updater-script not update-script
- "zip - . > X" instead of "zip X" otherwise zip(1) adds to an existing X

As mentioned on #5 certain microG images don't have any space on the /system and /vendor partitions; we can work around this by installing the overlay as a Magisk module. This is very simple to do, as shown here in this PR.

Tested on OnePlus 8T (KB2003), LineageOS 19.1 with MicroG.

### CODE REVIEW

**Comment by arovlad** [comment]
The whole point of this installation method is not using Magisk. Is there any point in creating a Magisk module when [one already exists](https://github.com/Androidacy/WebviewManager-Module)?

**Comment by infinity0** [comment]
@arovlad That other one is very heavy - it requires internet access and forces the user to download Bromite during the Magisk installation step. It's also unclear how Bromite is to be updated, whether it actually uses the user-installed copy or its own copy - its documentation says the user should re-install the module for updates. Also, that team produces good software but unfortunately due to lack of funding has decided to put all sorts of bloaty and ad-like features into their modules.

This module for  this PR is very light, it only includes your overlay, doesn't require internet access during installation, and the user can install Bromite separately however they want, and it is this copy that is used.

File: README.md
Comment (arovlad) [inline README.md:16]: This should be preserved as it is.

File: README.md
Comment (arovlad) [inline README.md:20]: Also mention that this is included for convenience only and only if you have Magisk already installed, as I personally do not endorse Magisk.

**Review by arovlad — CHANGES_REQUESTED** [review]
There are a couple of issues that need to be addressed before merging.

File: module.prop
Comment (infinity0) [reply module.prop]: sorry, fixed

File: README.md
Comment (infinity0) [reply README.md:5]: This paragraph is about the resource overlay, whereas *how to install* the overlay is a separate issue. So I think it's clearer to discuss what types of root access is required, in a different paragraph.

In fact, all installation methods require *some type* of root access. It is misleading from a security perspective to claim any of these methods "do not require [any type of] root access". More precisely, I think what you mean is that the custom recovery method means you don't require root access *on your main system*. This is fair, and I've amended the wording to say this, in a paragraph further below. But it is misleading to unconditionally say it "does not require root" with no other qualifiers - when you run `adb sideload` in custom recovery, this is effectively giving your script root access to my system, it *modifies the system partitions*.

File: module.prop
Comment (infinity0) [reply module.prop]: done, also bumped versionCode to match

**Review by arovlad — CHANGES_REQUESTED** [review]
There are a couple more changes that need to be done before merging.

**Comment by infinity0** [comment]
@arovlad For the sake of reaching an agreement and getting this PR merged, I have done what you asked. (The wording may be a little clumsy; if so please suggest more fixes.)

Separately I would like to comment that I strongly disagree with [this comment](https://www.reddit.com/r/LineageOS/comments/a4rlox/requesting_a_good_explanation_on_why_magisk_is_bad/ebj7zb6/) and I think it has an incorrect concept of security. While there is merit in keeping root access off your main system, it is weird and IMO a bit disingenious to complain that Magisk/MicroG is insecure simply because it can do horrible things to your system. All of these horrible things, **LineageOS itself can do**. That is the whole point of me rooting my own device, I become responsible for anything I install onto it. Assuming Magisk/MicroG/LineageOS themselves are secure, there is no way for an attacker to forcibly install bad stuff onto the user's device without their consent (or, it's a bug that must be fixed ASAP). So while there is merit in reducing the attack surface (so that it's only "LineageOS" rather than "LineageOS+MicroG+Magisk"), it is disingenuous to claim there is a fundamental security difference between them.

Similarly, I do not see a fundamental difference between "unlocking" vs "rooting" a device, IMO having an unlocked bootloader is already the same as having a "rooted device", so I hesitate to use this wording as you suggested.

On a normal unlocked device, the manufacturer is the God of your system, software updates must be signed by their key. On an unlocked device, whoever physically controls the device is the God of the device. Many people prefer the latter security model, which is why they unlock/root their device, generally technical people who are confident about their own security assessments. When I install a custom ROM I delegate some of my God powers to LineageOS, and likewise when I install Magisk/MicroG. A LineageOS dev complaining about Magisk/MicroG seems to me like someone who has been delegated God powers, complaining about other people also having been delegated God powers. I just can't buy into this logic.

**Review by arovlad — APPROVED** [review]

**Comment by arovlad** [comment]
You are arguments are very compelling indeed. 

I've done the README cleanup myself instead of requesting changes again because it's been two weeks and we are blocking the merge by discussing a tiny piece of documentation. 

If you do not agree with the changes please file an issue or create a new pull request. I will happily engage in another productive discussion. Thank you!

**Comment by infinity0** [comment]
Thanks for the merge. I'm happy with the final copywriting except for the wording "rooted device" as discussed above. I'm not sure how to convince you beyond what I already said above, but since you say you are happy to continue the discussion I'll file an issue about it.

**Pull request merged**
